instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/dict_util.py

```diff
@@ -115,6 +115,41 @@ def _flatten_with_flatten_func(
     return False, func_output
 
 
+def _flatten_compound_value_using_json(
+    key: str,
+    value: Any,
+    exclude_keys: Set[str],
+    rename_keys: Dict[str, str],
+    flatten_functions: Dict[str, FlattenFunc],
+    _from_json=False) -> FlattenedDict:
+    if _from_json:
+        _logger.debug(
+            "Cannot flatten value with key %s; value: %s", key, value
+        )
+        return {}
+    try:
+        json_string = json.dumps(value)
```

Comment:
btw this guy uses ascii by default, try setting some value to non-ascii and enjoy unreadable stuff in console and other places that don't utf8 it at rendering time

Reply:
I don't know that setting `ensure_ascii` to False is the right thing to do here. I believe that would create JSON that is not necessarily correct. And my understanding is that `json.loads` (which processes the string shortly thereafter) will perform the correct unescaping.

To double check this, I added a test to cover the JSON case, including one which uses a Unicode emoji character. 

I think I'd prefer to apply a "test driven development" approach on this one... let's come up with a test case that needs this to be changed. Otherwise, I think this is implemented correctly.

Reply:
Marking resolved.